### PR TITLE
Improve attrd client log messages

### PR DESCRIPTION
**Pull request by kgaillot**
This also separates XML schema-related code into its own source file.

### CODE REVIEW

File: crmd/control.c [hunk 1]
```diff
@@ -214,7 +214,7 @@ do_shutdown_req(long long action,
     xmlNode *msg = NULL;
 
     set_bit(fsa_input_register, R_SHUTDOWN);
-    crm_info("Sending shutdown request to %s", crm_str(fsa_our_dc));
+    crm_info("Sending shutdown request to all peers");
```
Comment (beekhof) [inline crmd/control.c]: Correct but it is interesting to know who we think the DC is/was

File: crmd/join_client.c [hunk 2]
```diff
@@ -226,7 +226,7 @@ do_cl_join_finalize_respond(long long action,
     }
 
     /* send our status section to the DC */
-    crm_debug("Confirming join join-%d: %s", join_id, crm_element_value(input->msg, F_CRM_TASK));
+    crm_debug("Confirming join join-%d", join_id);
```
Comment (beekhof) [inline crmd/join_client.c]: Reason? It is a debug message after all

File: crmd/control.c (hunk 1 above)
Comment (kgaillot) [inline crmd/control.c]: Good point

File: crmd/join_client.c (hunk 2 above)
Comment (kgaillot) [inline crmd/join_client.c]: It's always CRM_OP_JOIN_ACKNAK here, and "Confirming join join-1: ack_nack" isn't particularly helpful :-) Looking at it again, there is a more useful message printed immediately after this, so I think I'll get rid of this and improve the wording on the others.

File: lib/common/schemas.c [hunk 3]
```diff
@@ -717,10 +717,11 @@ update_validation(xmlNode **xml_blob, int *best, int max, gboolean transform,
     int next = -1;  /* -1 denotes "inactive" value */
 
     CRM_CHECK(best != NULL, return -EINVAL);
+    *best = 0;
+
     CRM_CHECK(xml_blob != NULL, return -EINVAL);
     CRM_CHECK(*xml_blob != NULL, return -EINVAL);
 
-    *best = 0;
```
Comment (jnpkrn) [inline lib/common/schemas.c]: good spot

**Pull request merged**